R/zchunk_LA100.0_LDS_preprocessing.R

```diff
@@ -8,58 +8,118 @@
 #' a vector of output names, or (if \code{command} is "MAKE") all
 #' the generated outputs: \code{object}. The corresponding file in the
 #' original data system was \code{LA100.0_LDS_preprocessing.R} (aglu level1).
-#' @details Describe in detail what this chunk does.
+#' @details Read in the various LDS datasets; regularize their column names and
+#' GLU data; change Taiwan ISO to that of mainland China; make LDS_ag_HA_ha and
+#' LDS_ag_prod_t tables consistent.
 #' @importFrom assertthat assert_that
 #' @importFrom dplyr filter mutate select
 #' @importFrom tidyr gather spread
-#' @author YourInitials CurrentMonthName 2017
-#' @export
-module_aglu_LA100.0_LDS_preprocessing_DISABLED <- function(command, ...) {
+#' @author BBL March 2017
+module_aglu_LA100.0_LDS_preprocessing <- function(command, ...) {
+
+  namelist <- c("Land_type_area_ha",
+                "LDS_ag_HA_ha",
+                "LDS_ag_prod_t",
+                "LDS_value_milUSD",
+                "MIRCA_irrHA_ha",
+                "MIRCA_rfdHA_ha",
+                "Mueller_yield_levels",
+                "Pot_veg_carbon_Mg_per_ha",
+                "Water_footprint_m3")
+  dirname <- "aglu/LDS/"
+
   if(command == driver.DECLARE_INPUTS) {
-    return(c( "i"))
+    x <- paste0(dirname, namelist)
+    names(x) <- rep("FILE", length(x))
+    return(x)
   } else if(command == driver.DECLARE_OUTPUTS) {
-    return(c("object"))
+    return(paste0("L100.", namelist))
   } else if(command == driver.MAKE) {
 
     all_data <- list(...)[[1]]
 
     # Load required inputs
-    i <- get_data(all_data, "i")
+    LDSfiles <- list()
+    for(nm in namelist) {
+      LDSfiles[[nm]] <- get_data(all_data, paste0(dirname, nm))
+    }
+
+    # Go through all data frames and...
+    for(nm in namelist) {
+
+      # Regularize data frame names
+      names(LDSfiles[[nm]]) %>%
+        sub("ctry_iso", "iso", .) %>%
+        sub("reglr_iso", "GTAP_region", .) %>%
+        sub("glu_code", aglu.GLU, .) %>%
+        sub("land_type", "land_code", .) %>%
+        sub("SAGE_crop", "GTAP_crop", .) %>%
+        sub("mirca_crop", "MIRCA_crop", .) %>%
+        sub("use_sector", "GTAP_use", .) ->
+        names(LDSfiles[[nm]])
+
+      # Replace numerical GLU code with concatenated GLU and the
+      # three-digit code (padded with zeroes as necessary)
+      if(aglu.GLU %in% names(LDSfiles[[nm]])){
+        LDSfiles[[nm]][[aglu.GLU]] <- paste(aglu.GLU, sprintf("%03d", LDSfiles[[nm]][[aglu.GLU]]), sep = aglu.GLU_NAME_DELIMITER)
+      }
+
+      # Re-set Taiwan to mainland China, as the current version of AgLU
+      # (and pre-2015 versions of FAOSTAT) doesn't disaggregate Taiwan
+      if("iso" %in% names(LDSfiles[[nm]])) {
+        d <- LDSfiles[[nm]]
+        if(nm != "Pot_veg_carbon_Mg_per_ha") {
```

Comment:
I guess I was asking from a scientific standpoint. Do we want Taiwan in that file? Does it not exist to begin with?